packages/flutter/test/widgets/image_test.dart

```diff
@@ -790,31 +788,34 @@ void main() {
       )
     );
 
-    // Two listeners - one is the listener added by precacheImage, the other by the ImageCache.
+    // Two listeners - one is the listener added by precacheImage, the other by the imageCache!.
     final List<ImageStreamListener> listeners = imageStreamCompleter.listeners.toList();
     expect(listeners.length, 2);
 
     // Make sure the first listener can be called re-entrantly
-    listeners[1].onImage(null, false);
-    listeners[1].onImage(null, false);
+    // TODO(darrenaustin): figure out how to handle this for NNBD
```

Comment:
@goderbauer any advice on how to handle this one? The `ImageListener` now requires a non-null `ImageInfo` as the first argument to `onImage()`. However, if I try the approach below it fails on line 803 with: 

```
The following assertion was thrown running a test:
'dart:ui/painting.dart': Failed assertion: line 1609: '<optimized out>': is not true.
```

Reply:
You probably have to clone the ImageInfo before passing it to the listeners since they all need their own image handle.

At least that's my guess based on the assert that's failing:

https://github.com/flutter/engine/blob/fd1ba9ccf05c474e490c1513e246710849286d19/lib/ui/painting.dart#L1609

/cc @dnfield

Reply:
Thx. If I clone the `ImageInfo` for each listener call, is that still testing what the original was testing?

Reply:
I think so, but @dnfield would know for sure.

Reply:
Correct, you need to clone the image info.

It's still valid for what this needs to test.

Reply:
Awesome. Thx.